#! /bin/sh -e
## 35_ip32-memsizeprint.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix RAM size output on ip32.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p0 ${patch_opts} < $0;;
    -unpatch) patch -R -p0 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

Index: arch/mips/sgi-ip32/ip32-memory.c
===================================================================
RCS file: /home/cvs/linux/arch/mips/sgi-ip32/ip32-memory.c,v
retrieving revision 1.1
diff -u -p -r1.1 ip32-memory.c
--- arch/mips/sgi-ip32/ip32-memory.c	12 Jan 2005 00:10:42 -0000	1.1
+++ arch/mips/sgi-ip32/ip32-memory.c	1 Feb 2005 00:32:14 -0000
@@ -36,8 +36,8 @@ void __init prom_meminit (void)
 		if (base + size > (256 << 20))
 			base += CRIME_HI_MEM_BASE;
 
-		printk("CRIME MC: bank %u base 0x%016lx size %luMB\n",
-			bank, base, size);
+		printk("CRIME MC: bank %u base 0x%016lx size %luMiB\n",
+			bank, base, size >> 20);
 		add_memory_region (base, size, BOOT_MEM_RAM);
 	}
 }
